src/DGtal/shapes/SurfaceMeshHelper.ih

```diff
@@ -0,0 +1,643 @@
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file SurfaceMeshHelper.ih
+ * @author Jacques-Olivier Lachaud (\c jacques-olivier.lachaud@univ-savoie.fr )
+ * Laboratory of Mathematics (CNRS, UMR 5127), University of Savoie, France
+ *
+ * @date 2020/02/18
+ *
+ * Implementation of inline methods defined in SurfaceMeshHelper.h
+ *
+ * This file is part of the DGtal library.
+ */
+
+
+//////////////////////////////////////////////////////////////////////////////
+#include <cstdlib>
+#include <limits>
+//////////////////////////////////////////////////////////////////////////////
+
+///////////////////////////////////////////////////////////////////////////////
+// IMPLEMENTATION of inline methods.
+///////////////////////////////////////////////////////////////////////////////
+
+///////////////////////////////////////////////////////////////////////////////
+// ----------------------- Standard services ------------------------------
+
+//-----------------------------------------------------------------------------
+template <typename TRealPoint, typename TRealVector>
+typename DGtal::SurfaceMeshHelper<TRealPoint, TRealVector>::SurfaceMesh 
+DGtal::SurfaceMeshHelper<TRealPoint, TRealVector>::
+makeSphere( Scalar radius, RealPoint center,
+            Size m, Size n, Normals normals )
+{
+  m = std::max( m, (Size)1 ); // nb latitudes (except poles)
+  n = std::max( n, (Size)3 ); // nb longitudes
+  // vertices are numbered as follows
+  //         0        : bottom pole
+  //         1 --    m: first row
+  //       m+1 --  2*m: second row
+  // ...
+  // (n-1)*m+1 --  n*m: before last row
+  //     n*m+1        : top pole    
+  const Size nbv = 2 + n * m;
+  std::vector< RealPoint > p( nbv );         // positions
+  std::vector< std::vector< Size > > faces; // faces = ( incident vertices )
+  p[     0 ] = center + radius * RealPoint( 0.0, 0.0, -1.0 );
+  p[ m*n+1 ] = center + radius * RealPoint( 0.0, 0.0,  1.0 );
+  const auto fv = [n] (Size i, Size j) { return i * n + j + (Size)1; };
+  for ( Size i = 0; i < m; ++i )
+    for ( Size j = 0; j < n; ++j )
+      {
+        const Scalar theta = ( 2.0 * M_PI * (double) j ) / (double) n;
+        const Scalar   phi = ( M_PI * (double) (i+1) ) / (double) (m+1) - 0.5 * M_PI;
+        p[ fv( i, j ) ] = center + radius * RealPoint( cos( theta ) * cos( phi ),
+                                                       sin( theta ) * cos( phi ),
+                                                       sin( phi ) );
+        // trace.info() << fv( i, j ) << "=(" << i << "," << j << ")"
```

Comment:
clean ? (perhaps we could add later a trace.debug())

Reply:
done